### ISSUE TITLE
Crash with Apple Pay and version 4.3.3 of Frames and 2.0.3 of Risk SDKs

### ISSUE THREAD

**Issue by kevinpalser** [checkout/frames-ios#526]
**Describe the bug**
We updated from 4.3.2 to 4.3.3. No problems previously. With 4.3.3 as soon as the Apple Pay biometric face matching step finished there is a crash. Looking at the crash log it took us to a log function in the Risk module being the last function call.

**To Reproduce**
1. Have custom integration, i.e not using the UI flow available from the SDK and calling the key functions directly.
2. Proceed to biometric confirmation step of Apple Pay and the charge. 

**Expected behavior**
Not to crash at the risk logging step.

**Screenshots**
Screenshot after tapping on the crash log to where the problem occurs.
<img width="1503" alt="Frames crash" src="https://github.com/checkout/frames-ios/assets/74354025/7f0b2671-4d48-4a1b-a8c7-94a64b610e95">

**Smartphone (please complete the following information):**
 - Device: iPhone 14 Pro Max 
 - OS: iOS 17.4.1

**Additional context**
The transaction when crashing went through to our server. This suggests the subsequent logging step is at fault. Seems to be replicable in that it keeps crashing. We reverted to Frames 4.3.2 & Risk 2.0.1 and the crash does not occur.

**Comment by okhan-okbay-cko** [comment]
Hey @kevinpalser, can you please share the crash log that is produced in an isolated project? 

Feel free to share it over here after you strip off the confidential/private information. 

Currently, we couldn't reproduce the issue.

**Comment by kevinpalser** [comment]
> Hey @kevinpalser, can you please share the crash log that is produced in an isolated project?
> 
> Feel free to share it over here after you strip off the confidential/private information.
> 
> Currently, we couldn't reproduce the issue.

Here is the crash report with semi confidential info marked as [***redacted***]

[2024-04-09_07-17-55.8156_+0100-16266a2d24b31400d7285b9d5fa32e7f08de0498.txt](https://github.com/checkout/frames-ios/files/14933127/2024-04-09_07-17-55.8156_%2B0100-16266a2d24b31400d7285b9d5fa32e7f08de0498.txt)

**Comment by okhan-okbay-cko** [comment]
Hey @kevinpalser, thanks for raising this and thanks for helping with additional data. 

Unfortunately, I can't rule out an exact reason for it now. Our sample apps all work as expected. Also, if I have a look at [the diff in between Risk SDK's 2.0.2 and 2.0.3 versions](https://github.com/checkout/checkout-risk-sdk-ios/compare/2.0.2...2.0.3), I can't see any culprits. 

If you'd like to help us further and isolate the issue in a sample project, please feel free.

Also, can you please share your Xcode version?

**Comment by kevinpalser** [comment]
> Unfortunately, I can't rule out an exact reason for it now. Our sample apps all work as expected. Also, if I have a look at [the diff in between Risk SDK's 2.0.2 and 2.0.3 versions](https://github.com/checkout/checkout-risk-sdk-ios/compare/2.0.2...2.0.3), I can't see any culprits.
Pitty - thanks for your efforts.
 
> If you'd like to help us further and isolate the issue in a sample project, please feel free.
It would need time to pull out and make a mini project. If we can allocate the time, we'll try.

> Also, can you please share your Xcode version?
Xcode 15.2 because of waiting on a resolution with another framework. I could test on Xcode 15.3.

**Comment by hai-ijji** [comment]
I'm pretty sure the Risk SDK 2.0.3 caused the issue. It crashed randomly for me, not only with Apple Pay but also with the new card form using Frames 4.3.3. I had to downgrade to Frames 4.3.2 and Risk 2.0.1 to fix the crash. 
Btw, I could never reproduce it in sandbox mode. It only crashed in Live mode, which we could only see if testing via TestFlight.

**Comment by kevinpalser** [comment]
Hi @hai-ijji 
> It only crashed in Live mode, which we could only see if testing via TestFlight. 

Same situation for us: live mode and TestFlight. Are you still using Frames 4.3.2 and Risk 2.0.1? We have not tried 4.3.4 because the change log reports minor version bump with no public facing changes.

**Comment by hai-ijji** [comment]
@kevinpalser Yeah, I think using Frames 4.3.2 and Risk 2.0.1 is fine for now. The changes in new Frames version were not significant and the changes in Risk SDK mostly for logs.

**Comment by kevinpalser** [comment]
@okhan-okbay-cko I'll try and make time tomorrow to test the 4.3.5 branch. Thanks.

**Comment by okhan-okbay-cko** [comment]
Thank you! You can use the tag directly if you prefer.

**Comment by kevinpalser** [comment]
@okhan-okbay-cko @hai-ijji 
I tested with the unreleased test branch with a live Apple Pay transaction using a TestFlight release and there was no crash.
![Captura de pantalla 2024-05-01 a les 16 48 33](https://github.com/checkout/frames-ios/assets/74354025/ca037ab9-84e5-42c1-916f-f6f8f95b46e3)

**Comment by okhan-okbay-cko** [comment]
I'll be closing this issue. Please feel free to re-open or create a new one if you face with the same crash.

**Comment by woolstrand** [comment]
This still happens for us. 4.3.5 crashed in live environment immedately after successful face recognition. Attaching crash log which might probably help.
[2024-05-08_15-29-40.8635_+0200-9b1d793c0dd6bf52362a832dd694d8c233b2b03b.crash.txt](https://github.com/checkout/frames-ios/files/15267215/2024-05-08_15-29-40.8635_%2B0200-9b1d793c0dd6bf52362a832dd694d8c233b2b03b.crash.txt)

**Comment by kevinpalser** [comment]
@okhan-okbay-cko unfortunately like @woolstrand we also had a couple of crashes yesterday recorded yesterday with 4.3.5 and shall revert back to 4.3.2. The 4.3.5 crash frequency is less than 4.3.3/4 but still less stable than 4.3.2.

**Comment by Prakaship78** [comment]
I am still able to face this crash issue in 4.3.5 version. Downgrading to 4.3.2 is working fine .
I am not using any face recognition , as soon as the 3DSWebViewController is presented app is getting crashed .
<img width="1159" alt="image (6)" src="https://github.com/checkout/frames-ios/assets/43235948/f48ebe5c-68a2-4583-b490-3880edee690e">

**Comment by jheng-hao-lin-cko** [comment]
Hi everyone, our apologies for causing all inconvenience. We are aware of this issue and actively working on it. Will keep you posted on this issue.

**Comment by okhan-okbay-cko** [comment]
Hey all, please give version `4.3.6` a go and let us know if you still are having issues. Thanks.

**Comment by kevinpalser** [comment]
@woolstrand @Prakaship78 @hai-ijji 
Did you try 4.3.6?

Our company has been weary to update because unlike 4.3.3, which crashed immediately, 4.3.5 crashes were rarer making it difficult to gauge and less easily replicable. So we didn't want to risk live crashes with 4.3.6 based on our 4.3.5 experience even on a slow roll out.

**Comment by Prakaship78** [comment]
Yes @kevinpalser  did a proper round of testing on sanbox only and app is not crashing anymore .Need to test on prod after sometime and will let your know .
Thanks for the quick fix
